Partial-CI-Tests-Structure.md

```diff
@@ -0,0 +1,22 @@
+## Introduction
+
+Our CI process has a workflow, `check_ci_test_suites_to_run`, that determines which test suites to run for a given PR based on the files that the PR modified.
+If your PR fails due to the `check_ci_test_suites_to_run` workflow, please look at the error log and depending on the error, do the following:
+
+* If you see an error about the root files config, then see the [root files config section](#root-files-config) below.
+
+## Overview of Partial CI Tests Structure
+
+For non-python files, we run partial ci tests depending on the tests that a specific file affects. Running partial tests like this allow for PRs to be checked faster on the tests that a PR actually modifies. This faster test speed allows developers to get feedback faster and allows CI resources to be free more. The process of determining which tests to run goes like this:
+
+1. Generate a root files mapping which maps files to their root files. This file is generated at the start of every PR check run and is stored at `root-files-mapping.json`.
+2. Store golden files containing the page modules that a specific test uses. These golden files are stored at `core/tests/test-modules-mappings/{{test-type}}/{{test-name}}`.
```

Comment:
This is actually already part of the codebase, right? If so, mention it in the paragraph above this list (describe these files in the codebase and what they represent) before describing the process, and only have steps 1 and 3 as part of the "process".

Reply:
Well the PR hasn't been merged yet for the test-modules-mappings and the root-files-mapping is generated for every PR.

Reply:
Oh, my comment was taking into account that you'd merge the other PR before this one. Could you please read it again with that in mind?

Reply:
So are you suggesting that I move the 2nd bullet point out (and describe in more in depth with examples)? Though @U8NWXD commented that I should merge a wiki page for Partial-CI-Tests-Structure beforehand, so I suppose I can give references to files that will exist?

Reply:
Let's have your wiki page refer to the correct eventual locations of files. If a file doesn't exist yet and there's going to be a bit of a gap before it does, add a note to say that it'll be introduced in PR XXX and file an issue on oppia-web-developer-docs to discard the note later. Otherwise it's probably OK to have an inconsistency for a short time (a couple of days), or wait a couple of days to merge the wiki PR.

Thinking about the ideal page, I was pointing out that the process that happens for each PR seems to constitute 1 and 3 only, because 2 is already part of the codebase -- so 2 doesn't really belong with the process that is executed for each PR, which consists of 1 and 3 only. (Or am I misunderstanding how the process works?)

Reply:
Done.